tt_metal/impl/device/device_pool.cpp

```diff
@@ -225,13 +222,15 @@ void DevicePool::init_firmware_on_active_devices() const {
         auto tunnels_from_mmio = tt::Cluster::instance().get_tunnels_from_mmio_device(mmio_device_id);
         this->initialize_device(dev);
         detail::InitDeviceProfiler(dev);
-        for (uint32_t t = 0; t < tunnels_from_mmio.size(); t++) {
-            // Need to create devices from farthest to the closest.
-            for (uint32_t ts = tunnels_from_mmio[t].size() - 1; ts > 0; ts--) {
-                uint32_t mmio_controlled_device_id = tunnels_from_mmio[t][ts];
-                log_debug(tt::LogMetal, "Tunnel {} Device {} Tunnel Stop: {}", t, mmio_controlled_device_id, ts);
-                this->initialize_device(this->devices[mmio_controlled_device_id].get());
-                detail::InitDeviceProfiler(this->devices[mmio_controlled_device_id].get());
+        if (not this->skip_remote_devices) {
```

Comment:
under what conditions are we going to skip remote devices?

Reply:
Mostly for CI jobs that run on N300. There are a lot of single chip tests in there, without skipping remote devices, the pipelines take twice as long to finish.

Reply:
It's only used in `CreateDevice(0, skip_remote=true)`